### PR TITLE
feat: add support for variable sleep durations

### PR DESCRIPTION
**Pull request by pimlie**
This PR adds support to configure different sleep durations per day-of-week, hour and minute. It should basically support the most common configs as is described in [hass.md](../blob/main/hass.md), so instead of using home-assistant to keep changing the sleep duration it's just baked into the homeplate fw :)

- see `config_example.cpp` for example config
- I had to change the env's a little bit to be able to run the unit tests

### CODE REVIEW

File: src/config_example.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
+// NOTE: THIS FILE IS ONLY NEEDED WHEN CONFIGURING VARIABLE SLEEP
+// DURATIONS USING TIME BLOCKS. IF USING A SINGLE SLEEP DURATION THEN
+// THIS FILE CAN BE OMITTED
+// #include "config.h"
+
+// How long to sleep between image refreshes
+// - there is no validation of any kind, make sure your blocks are continuous
+// - time blocks should be configured per day
+// - dow = DayOfWeek, starts at 1 = Monday to 7 = Sunday
+// SleepTimeBlock sleepTimeBlocks[] = {
```
[the remark this answers is not in the document]
Comment (lanrat) [reply src/config_example.cpp:10]: I do like `sleepIntervals` more than `sleepTimeBlocks`. I also like `sleepSchedule`.

File: README.md [hunk 2]
```diff
@@ -43,9 +43,39 @@ Setup [sibbl](https://github.com/sibbl/)'s [hass-lovelace-kindle-screensaver](ht
...
+#### Configuring
+
+Copy `config_example.h` to `config.h` and add/change your settings.
+
+##### Variable sleep intervals
+
+If you want your inkplate to sleep with different intervals, copy `config_example.cpp` to `config.cpp` and uncomment the 3 lines in `config.h` starting from `#include "sleep_duration.h"`. Then configure your sleepTimeBlocks.
+
+Note that time blocks do not span multiple days, this means that the *day of week* setting is similar to configuring a cronjob. F.e. the settings below should be read as *between Xam to Ypm on every weekday*, and **not** as *from monday Xam to friday Ypm*.
+```cpp
```
Comment (lanrat) [inline README.md:57]: there should be a newline after the text before the code block

File: README.md [hunk 3]
```diff
@@ -99,3 +129,16 @@ Waveform load failed! Upload new waveform in EEPROM. Using default waveform.
 ```
 
 Older Inkplates don't appear to ship with an updated waveform. I found waveform 3 looks the best for mine.
+
+### Tests
+
+The available unit tests use the 'native' environment and can be run by either:
+
+- running them manually
```
Comment (lanrat) [inline README.md:137]: Use `*` for unordered list to stay consistent with the rest of the file

File: README.md [hunk 4]
```diff
@@ -99,3 +129,16 @@ Waveform load failed! Upload new waveform in EEPROM. Using default waveform.
...
+
+```shell
+pio test -v
+```
+
+- in VSCode use Testing -> native -> Run Test
```
Comment (lanrat) [inline README.md:143]: Same as above, use `*` for lists.

**Review by lanrat — CHANGES_REQUESTED** [review]
Thanks for the new feature! I left a few very minor nits.

**Review by lanrat — APPROVED** [review]

**Comment by lanrat** [comment]
Thanks for the PR and new feature!

**Comment by felixh10r** [comment]
@pimlie @lanrat a quick heads-up because it took me a minute to find the problem: in `config_example.h`, `sleep_duration.h` was [renamed](https://github.com/lanrat/homeplate/commit/323ec59266118be3aac48873c3d89cf6640d3ef0#diff-ec67fd79bb9800179eb8f05715869f3faa7e37dc8e717a369cbd4983a986b446R21) to `sleep_schedule.h` but the file wasn't renamed. this line should be reverted, otherwise the correct file can't be found on build (i thought it's more elegant to amend the commit then to open a new PR—if i should, just say the word!)

**Pull request merged**